### PR TITLE
Upgrade InjectCSS sample from SPFx v1.8.0 to v1.20.0

### PR DESCRIPTION
**Pull request by tom-daly**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/sp-dev-fx-extensions/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                              |
| New feature?    | no                             |
| New sample?     | no                               |
| Related issues? | #1414 |

## What's in this Pull Request?

These changes upgrade the solution from SPFx v1.8.0 to v1.20.0. 

It also fixes the warning from the linter by adding the proper typing

It also includes modified instructions on how to deploy using PnP PowerShell since the new instructions Sept 9, 2024.

### CODE REVIEW

**Comment by Adam-it** [comment]
Thank you for opening a PR to our samples repo and helping us upgrade our project 👍
We will try to review this PR ASAP

**Review by Adam-it — CHANGES_REQUESTED** [review]
@tom-daly Awesome work 👏👏👏
I had a quick check and managed to build and deploy the solution on my tenant. You Rock 🤩

I noticed a few old dependencies which should be aligned with SPFx version. I know the extension does not use them but since they are in the packge.json file we may fix them up as well. Please do give them a double check before we proceed 🙏

this kind of info you may always catch using the `validate` action using SPFx Toolkit or `m365 spfx project doctor` command using CLI for Microsoft 365.

**Comment by tom-daly** [comment]
This was the only 1 of my commits where I used the m365 cli to check the upgrade and it didn't work well. I also ran the dr and it did not pick up your suggestions but just remove gulp and add gulp-cli. I've taken the package dependencies from a clean v1.20.0 project and will be checking that in momentarily.

**Comment by Adam-it** [comment]
> This was the only 1 of my commits where I used the m365 cli to check the upgrade and it didn't work well. I also ran the dr and it did not pick up your suggestions but just remove gulp and add gulp-cli. I've taken the package dependencies from a clean v1.20.0 project and will be checking that in momentarily.

I used SPFx Toolkit to support me in this review and the validate action picked up this as well. 
if you had any problems with CLI for M365 please (if you have time) open up an issue in the CLI repo so that we may pick it up.
Anyway thanks for a quick turn around. I will do a recheck ASAP

File: package-lock.json
Comment (Adam-it) [inline package-lock.json:1]: I think we should not update the package-lock.json which is outside of the sample folder. 
Lets revert this

File: samples/react-application-injectcss/build.cmd
Comment (Adam-it) [inline samples/react-application-injectcss/build.cmd:1]: I now it's not a blocking comment but in general I think that it this is not included in the guidance as preferable way of building I would avoid including this as it is windows specific and seems more like work preference method then something specific to the sample

**Review by Adam-it — CHANGES_REQUESTED** [review]
@tom-daly thanks for the quick turn around. 
I think we are almost there. Let's do a bit of clean up and we should be ready to merge 🚀

File: samples/react-application-injectcss/build.cmd
Comment (tom-daly) [reply samples/react-application-injectcss/build.cmd:1]: its a personal preference removed for the examples, removing file

**Comment by Adam-it** [comment]
@tom-daly are you still working on this PR? If it is ready to be merged please do let us know by hitting the `Ready for review` button 👍

**Comment by Adam-it** [comment]
@tom-daly Awesome work 👏👏👏👏 
You Rock 🤩

**Pull request merged**